src/query/service/src/sql/planner/plans/scalar.rs

```diff
@@ -511,7 +511,7 @@ impl ScalarExpr for CastExpr {
     }
 
     fn is_deterministic(&self) -> bool {
-        true
+        false
```

Comment:
`Cast` should be deterministic.

`random` is not deterministic, so `cast(random())` is not deterministic.  We should find the root reason.

Reply:
should be `self.argument.is_deterministic()`

Reply:
ok